src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets

```diff
@@ -252,6 +252,7 @@
     <!-- Ensure output directories. -->
     <MakeDir Directories="%(_Protobuf_OutOfDateProto.OutputDir)" />
     <MakeDir Directories="%(_Protobuf_OutOfDateProto.GrpcOutputDir)" />
+    <MakeDir Directories="%(Protobuf_ExpectedOutputs.RelativeDir)" />
```

Comment:
Interesting trick, can you elaborate a bit more on how this works and why it works reliably? (especially in cornercases).

Reply:
I was trying to find out how we can make sure, the generated paths exist. I looked into the MSBuild code of `ProtobufCompilerOutputs` and saw, that the information about the path is already implicitly available in `PossibleOutputs` which gets assigned to `$(Protobuf_ExpectedOutputs)`. These are already prepared to be located in `%(_Protobuf_OutOfDateProto.OutputDir)` (as this will be the input to `ProtobufCompilerOutputs` a couple lines later) and contain the generated directory hash.

`%(Protobuf_ExpectedOutputs.RelativeDir)` therefore is the union of all paths that will be generated (independent of what really is part of `_Protobuf_OutOfDateProto`, but as all those paths are necessary already for the first creation, i found that ignorable also for big sets of protos).

We might be able to improve this by restrict it like `Condition = " '%(Source)' != '' and '@(Protobuf_ExpectedOutputs)' == '' "` (as done with the definition of `_Protobuf_OutOfDateProto` in `_Protobuf_GatherStaleSimple`) but I'm not too deep into MSBuild to know if this does what I think it does. As `%(Protobuf_ExpectedOutputs.Source)` is pointing to the proto of the file this expected output is related to, this might work.

Reply:
I made it more explicit by introducing `_Protobuf_ExpectedGenerated` as junction of `Protobuf_ExpectedOutputs` and `_Protobuf_OutOfDateProto`. `%(_Protobuf_ExpectedGenerated.RelativeDir)` is therefore the union of all directories of expected files that have been modified.